src/Thesis/LaplaceNoise.hs

```diff
@@ -1,17 +1,18 @@
 -- | Generates Laplace noise
 module Thesis.LaplaceNoise where
 
-import System.Random (StdGen, random)
+import System.Random (StdGen, random, setStdGen)
 
 -- | Generates a random variable drawn from a Laplace distribution with the given scale using the
 -- given random number generator.
-generate :: StdGen -> Double -> (Double, StdGen)
-generate gen scale =
+generate :: StdGen -> Double -> IO Double
```

Comment:
I don't think the type of the generate function has to change, the caller can decide how to handle the seed.

Also, the current implementation is somewhat ambivalent. It takes an stdgen, but then it stores the new stdgen in the global IO monad. If it were to work in the IO monad, it could then just take the stdgen from the global environment.